screens/SignIn.tsx

```diff
@@ -1,220 +1,64 @@
-  };
-  const postRequest = (result) => {
-    fetch("https://resell-dev.cornellappdev.com/api/auth/login/", {
-      method: "POST",
-      headers: {
-        Accept: "application/json",
-        "Content-Type": "application/json",
-      },
-      body: JSON.stringify(result),
-    })
-      .then((response) => response.json())
-      .then(async (json) => {
-        if (json.error) {
-          alert("Please Use Your Cornell Email :)");
-        } else {
-          if (!onBoard) {
-            const myUserRef = await userRef.doc(result.user.email);
-            const doc = await myUserRef.get();
-            if (doc.exists) {
-              console.log(doc.data());
-              setOnBoarded(doc.data().onboarded);
-            }
-          }
-          storeAccessToken(json.accessToken);
-          storeUserId(json.userId);
-          storeEmail(result.user.email);
-          storeExpireAt(json.expiresAt + "");
-          storeRefreshToken(json.refreshToken);
-          register(result.user.email, json.userId);
-        }
-      })
-      .catch((error) => {
-        console.error(error);
-      });
-  };
-
-  const storeAuthUser = async (user, userId) => {
+  const [user, setUser] = useState<User>(null);
+  const [onboard, setOnboard] = useState(false);
+  const signIn = async () => {
     try {
-      const response = await fetch(
-        "https://resell-dev.cornellappdev.com/api/user/id/" + userId
-      );
-      if (response.ok) {
-        const json = await response.json();
-        user
-          .updateProfile({
-            displayName: json.user.givenName + " " + json.user.familyName,
-            photoURL: json.user.photoUrl,
-          })
-          .then(() => {
-            console.log("finish");
-            log_in(accessToken);
-
-            storeSignedIn("true");
-            setSigned(true);
-          })
-          .catch((error) => {
-            alert(error);
-          });
+      // Not really sure why we need this line but the guide set to put it in
+      // Maybe it's required for Android? If not, remove it.
+      await GoogleSignin.hasPlayServices();
+      const userInfo = await GoogleSignin.signIn({});
+      const mUser = userInfo.user;
+      if (mUser != null) {
+        setUser(userInfo);
+        const googleCredential = auth.GoogleAuthProvider.credential(
+          userInfo.idToken
+        );
+        return auth().signInWithCredential(googleCredential);
       }
+
+      console.log(`user info = ${JSON.stringify(mUser)}`);
     } catch (error) {
-      console.error(error);
+      if (error.code === statusCodes.SIGN_IN_CANCELLED) {
+        console.log("Sign in was cancelled");
+      } else if (error.code === statusCodes.IN_PROGRESS) {
+        console.log("operation already in progress");
+      } else if (error.code === statusCodes.PLAY_SERVICES_NOT_AVAILABLE) {
+        console.log("play services not available");
+      } else {
+        console.log(`Some other error ${error}`);
+      }
```

Comment:
It'd probably be cleaner to replace this with a switch statement on `error.code`